sda-download/api/sda/sda.go

```diff
@@ -207,12 +207,22 @@ func Files(c *gin.Context) {
 
 // Download serves file contents as bytes
 func Download(c *gin.Context) {
-	if c.Param("type") != "encrypted" && !config.Config.App.ServeUnencryptedData {
+	// This conditional should always be satisfied for /s3 since the c.Param is set to  encrypted
+	// when Crypt4GHPublicKeyB64 is not set or empty, but this is not the case for calls to /files endpoint.
+	// So we need this check.
+
+	if c.Param("type") != "encrypted" && config.Config.App.Crypt4GHPublicKeyB64 == "" {
 		c.String(http.StatusBadRequest, "downloading unencrypted data is not supported")
 
 		return
 	}
 
+	if c.Param("type") != "encrypted" && c.GetHeader("Client-Public-Key") != "" {
```

Comment:
I'm not following...
if `type=decrypted` (or at least `!=encrypted`), and there is no client public key provided,  then the user has requested un-encrypted data. Why should we give a bad request with  `downloading encrypted data is not supported`?

Reply:
Yeap, this can be confusing:
If `type!=encrypted` then this means that `download` is deployed as serving unencrypted files (see [here](https://github.com/neicnordic/sensitive-data-archive/blob/f1979be39c23318500ba58cb340608689431557c/sda-download/api/s3/s3.go#L250) for example). 

But if the user has provided a Client pub key in the header that means that the user is trying to download encrypted files (i.e. encrypted with their provided pubkey).
Thus, this will throw a bad request error (as per the issue description).

Reply:
ah, I thought the `type` came from the user's request. But in that case, it makes much sense :+1: .